scripts/prepare-subcharts

```diff
@@ -117,25 +117,22 @@ while ! [[ -z $(yq r ${f}/charts/requirements.yaml "dependencies[${i}]") ]]; do
 		((i=i+1))
 		continue
 	fi
-	subchart_path=$(ls ${f}/charts/charts/${subchart_tgz})
-	if [[ ${version} == *'*'* ]]; then
+	if [[ ! ${version} =~ ^[0-9\.]+$ ]]; then
 		# Subcharts that are tracked remotely must have a fixed version number
-		if [[ -f ${subchart_path} ]]; then
-			rm -rf ${charts_path}
-		fi
+		rm -rf ${charts_path}
 		((i=i+1))
 		continue
 	fi
 	# Subchart could be patched
-	if [[ -f ${subchart_path} ]]; then
+	if [[ -f ${f}/charts/charts/${subchart_tgz} ]]; then
 		tar xvzf ${f}/charts/charts/${subchart_tgz} -C ${f}/charts/charts > /dev/null 2>&1
 		rm ${f}/charts/charts/${subchart_tgz}
 	fi
 	((i=i+1))
 done
 
 # Undo the patch on the requirements.yaml
-if [[ -f ${f}/requirements.patch ]] && [[ -s ${f}/requirements.patch ]]; then
-	patch -R -E -p3 -d ${f}/charts < ${f}/requirements.patch
+if [[ -f ${f}/requirements.patch ]]; then
+	[[ -s ${f}/requirements.patch ]] && patch -R -E -p3 -d ${f}/charts < ${f}/requirements.patch
```

Comment:
feel like this got more confusing, why'd it move ?

Reply:
We want to delete the `requirements.patch` if it's an empty file but only patch it if it actually has contents.

Without this, an empty `requirements.patch` would be left behind.

I'll update it in the next commit to make that more clear.

Reply:
Yes that reads cleaner now, thanks.